tests/test_datasets/test_datasets/test_face_datasets/test_face_300vw_dataset.py

```diff
@@ -0,0 +1,182 @@
+            data_mode='topdown', test_mode=True)
+        self.assertEqual(dataset.data_mode, 'topdown')
+        self.assertEqual(dataset.bbox_file, None)
+        self.assertEqual(len(dataset), 3)
+        self.check_data_info_keys(dataset[0])
+
+    def test_bottomup(self):
+        # test bottomup training
+        dataset = self.build_face_300vw_dataset(data_mode='bottomup')
+        self.assertEqual(len(dataset), 3)
+        self.check_data_info_keys(dataset[0], data_mode='bottomup')
+
+        # test bottomup testing
+        dataset = self.build_face_300vw_dataset(
+            data_mode='bottomup', test_mode=True)
+        self.assertEqual(len(dataset), 3)
+        self.check_data_info_keys(dataset[0], data_mode='bottomup')
+
+    def test_exceptions_and_warnings(self):
+
+        with self.assertRaisesRegex(ValueError, 'got invalid data_mode'):
+            _ = self.build_face_300vw_dataset(data_mode='invalid')
+
+        with self.assertRaisesRegex(
+                ValueError,
+                '"bbox_file" is only supported when `test_mode==True`'):
+            _ = self.build_face_300vw_dataset(
+                data_mode='topdown',
+                test_mode=False,
+                bbox_file='temp_bbox_file.json')
+
+        with self.assertRaisesRegex(
+                ValueError, '"bbox_file" is only supported in topdown mode'):
+            _ = self.build_face_300vw_dataset(
+                data_mode='bottomup',
+                test_mode=True,
+                bbox_file='temp_bbox_file.json')
+
+        with self.assertRaisesRegex(
+                ValueError,
+                '"bbox_score_thr" is only supported in topdown mode'):
+            _ = self.build_face_300vw_dataset(
+                data_mode='bottomup',
+                test_mode=True,
+                filter_cfg=dict(bbox_score_thr=0.3))
+
+    def test_annotation(self):
+        dataset = self.build_face_300vw_dataset(
+            data_mode='topdown', test_mode=True)
+
+        meta = {
+            'keypoint_id2name': None,
+            'keypoint_name2id': None,
+            'keypoint_colors': [(255, 255, 255)],
+            'skeleton_links': None,
+            'skeleton_link_colors': None
+        }
+        visualizer = FastVisualizer(meta, radius=1)
+
+        vis_dir = os.path.join(os.path.dirname(dataset.ann_file), 'vis')
+        os.makedirs(vis_dir, exist_ok=True)
+
+        for i in range(len(dataset)):
+            instance = dataset.prepare_data(i)
+            assert instance['keypoints'].shape == (
+                1, 68, 2), '300VW keypoints shape should be (1, 68, 2)'
+            img = cv2.imread(instance['img_path'])
+            visualizer.draw_points(img, instance['keypoints'][0])
+            save_path = os.path.join(vis_dir,
+                                     os.path.basename(instance['img_path']))
+            assert cv2.imwrite(save_path,
+                               img), 'facial keypoints visualization failed!'
```

Comment:
If you want to test the visualizer, you can add test code here https://github.com/open-mmlab/mmpose/blob/main/tests/test_visualization/test_fast_visualizer.py. And use a temporary path instead of a real path.